### DISCUSSION TITLE
Creating new note from open pdf

### DISCUSSION THREAD

**Discussion by pwolper** [org-roam/org-roam-bibtex#274] [Q&A]
This night be a silly question related to a specific task, bit I am wondering if i can create the org-roam note for my citation from the buffer I am viewing the pdf in, when no note exists yet. I often find myself reading a pdf already beofre I decide to take a note, and have to navigate the helm or ivy interface again to  creat the new note. Is there an easy way to do this directly from the open buffer of the pdf?

**Comment by myshevchuk** [comment]
The short answer is no, this is not possible currently. A longer answer is I could implement it with some prerequisites/restrictions.

Case 1: The pdf filename without extension contains or matches entry's citation key, e.g. `author2023book.pdf` or `author2023book chapter 1.pdf`. This should be pretty straightforward and robust.
Case 2: The pdf file name is an arbitrary string, which is stored in the `file` field of the BibTeX entry, e.g.
`file = {/home/user/library/arbitrary file name.pdf}`.  This would also be easy but in theory is error-prone because one could have `arbitrary file name.pdf` stored under several entries, which however is rather unlikely, so it will probably yield the desired result in most cases.

These two cases should probably cover more than 95% of use cases. What will definitely not be implemented is trying to create a note for a pdf file that does not have a corresponding BibTeX entry because 1) ORB relies on BibTeX database to retrieve the relevant info; 2)  ORB is not a reference manager, so it does not modify the BibTeX database, for example, to add new entries from a pdf file.

If your use case falls within the above description then please feel free [to file a new feature request](https://github.com/org-roam/org-roam-bibtex/issues/new?assignees=&labels=&projects=&template=feature_request.md&title=).
